src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -669,20 +695,20 @@ Exception GetInvalidArgumentsException()
             }
         }
 
-        private void ThrowInvalidArguments(int offset, int length)
+        private void ThrowInvalidArguments(int offset, int length, ExceptionArgument offsetOrStart = ExceptionArgument.offset)
```

Comment:
Avoid using default arguments for internal helper methods like this. Since it impacts user-visible behavior, it's important for the callers to be clear about what explicit argument is being passed in here. It also helps the callers see that _four_ arguments (including the implicit this) - not three - are now being passed to this method, which affects codegen.